kimchi/src/circuits/polynomials/keccak/mod.rs

```diff
@@ -98,6 +103,82 @@ pub(crate) fn expand_word<F: PrimeField, T: ExprOps<F>>(word: u64) -> Vec<T> {
         .collect::<Vec<T>>()
 }
 
+/// Pads the message with the 10*1 rule until reaching a length that is a multiple of the rate
+pub(crate) fn pad(message: &[u8]) -> Vec<u8> {
+    let mut padded = message.to_vec();
+    padded.push(0x01);
+    while padded.len() % 136 != 0 {
```

Comment:
The code looks very imperative. I don't know which style rustaceans prefer to code in: functional or imperative?
As a functional programmer, I tend to prefer the functional style. I would then use an approach without any side effect on the vector padded. `padded.len()` has a different value every run as we push, which makes it not "functional friendly".